### PR TITLE
Ledger.info api

### PR DESCRIPTION
**Pull request by tantommy**
- modules under `/network/modules` are mixins and name-spaced via the `_namespace_` property on the mixin
- changed from `ledger.fetchLedgerInfo()` to `ledger.info()`
- symbols are optional in `ledger.balance()`
- adds better type support for modules when applied to Network

### CODE REVIEW

**Comment by hansl** [comment]
Is this ready to review?

**Comment by tantommy** [comment]
> Is this ready to review?

it is ready now @hansl .  thanks!

File: src/network/modules/ledger/index.ts [hunk 1]
```diff
@@ -0,0 +1,69 @@
+import cbor from "cbor";
+import { Identity } from "../../../identity";
+import { Message } from "../../../message";
+
+export interface LedgerInfo {
+  symbols: Map<ReturnType<Identity["toString"]>, string>;
+}
+
+export class Ledger {
+  async fetchLedgerInfo(): Promise<LedgerInfo> {
```
Comment (hansl) [inline src/network/modules/ledger/index.ts:10]: Call this `info()`, like the `ledger.info()`.

**Comment by tantommy** [comment]
@hansl this is now ready for review

File: src/network/network.ts [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+  [k: string]: any
+  url: string
+  keys: KeyPair | undefined
+
+  constructor(url: string, keys?: KeyPair) {
+    this.url = url
+    this.keys = keys
+  }
+
+  apply(modules: NetworkModule[]) {
```
Comment (hansl) [inline src/network/network.ts:18]: You should make this type safe as early as possible to avoid issues later on.

**Review by hansl — APPROVED** [review]

**Pull request merged**